infra/src/main/kotlin/fi/fta/geoviite/infra/configuration/RequestFilter.kt

```diff
@@ -137,11 +159,31 @@ class RequestFilter @Autowired constructor(
             val content = getJwtData(request)
             log.info("JWT authorization headers processed: " +
                     "validated=$validationEnabled user=${content.userDetails.userName} groups=${content.groupNames}")
-            val role = authorizationService.getRoleByUserGroups(content.groupNames) ?: throw ApiUnauthorizedException(
+
+            val availableRoles = authorizationService.getRolesByUserGroups(content.groupNames) ?: throw ApiUnauthorizedException(
                 "User doesn't have a valid role: userId=${content.userDetails.userName} groups=${content.groupNames}"
             )
-            User(content.userDetails, role)
+
+            User(
+                details = content.userDetails,
+                role = authorizationService.getDefaultRole(availableRoles),
+                availableRoles = availableRoles
+            )
+        }
+    }
+
+    private fun getActiveUserRole(request: HttpServletRequest, user: User): Role {
+        return request.cookies?.firstOrNull { cookie ->
+            cookie?.name == DESIRED_ROLE_COOKIE_NAME
+        }
+        ?.let { desiredRoleCookie ->
+            val desiredRoleCode = Code(desiredRoleCookie.value)
+
+            user.availableRoles.find { availableRole ->
+                availableRole.code == desiredRoleCode
+            }
         }
+        ?: authorizationService.getDefaultRole(user.availableRoles)
```

Comment:
Rupesin miettimään mikähän olisi paras tapa hanskata tilanne jossa keksin haluamaa roolia ei ole 🤔 Toisaalta saattaisi olla fiksua vaan heittää defaulttirooliin defaulttaamisen sijasta (ehkä käyttäjä yritti jotain outoa -> seinä pystyyn,) mutta toisaalta meillä saattaa olla tilanteita joissa käyttäjän ainoa rooli muuttuu vaikka konsultista virastokäyttäjäksi tai toisin päin, ja olisi kiva jos Geoviite osaisi hanskata ne tilanteet kivemmin kuin nostamalla tuo sama seinä pystyyn. Tuo nykylogiikkahan taas puolestaan hanskaa ne ihan kivasti. Onko ihmisillä mielipiteitä?

Reply:
Tuo headerihan ei tosiaan ole mikään tietoturvajuttu vaan vain käyttäjän preferenssi ja Code tosiaan jo varmistaa että on se ainakin vähän oikeahkon näköinen string. Eli sanoisin että on ihan OK että puuttuvan roolin tapauksessa käyttäydytään samoin kun jos keksiä ei olisi ollenkaan eli poimitaan default.